Hippo.Web/ClientApp/src/components/ClusterAdmin/Clusters.tsx

```diff
@@ -0,0 +1,324 @@
+                  cluster: {
+                    ...editClusterModel.cluster,
+                    description: e.target.value,
+                  },
+                };
+                setEditClusterModel(model);
+                setReturn(model);
+              }}
+            />
+          </div>
+          <div className="form-group">
+            <label htmlFor="fieldSshUrl">SSH Url</label>
+            <input
+              className="form-control"
+              id="fieldSshUrl"
+              value={editClusterModel.cluster.sshUrl}
+              onChange={(e) => {
+                const model: ClusterModel = {
+                  ...editClusterModel,
+                  cluster: {
+                    ...editClusterModel.cluster,
+                    sshUrl: e.target.value,
+                  },
+                };
+                setEditClusterModel(model);
+                setReturn(model);
+              }}
+            />
+          </div>
+          <div className="form-group">
+            <label htmlFor="fieldSshName">SSH Name</label>
+            <input
+              className="form-control"
+              id="fieldSshName"
+              value={editClusterModel.cluster.sshName}
+              onChange={(e) => {
+                const model: ClusterModel = {
+                  ...editClusterModel,
+                  cluster: {
+                    ...editClusterModel.cluster,
+                    sshName: e.target.value,
+                  },
+                };
+                setEditClusterModel(model);
+                setReturn(model);
+              }}
+            />
+          </div>
+          <div className="form-group">
+            <label htmlFor="fieldSshKey">SSH Key</label>
+            <textarea
+              className="form-control"
+              id="fieldSshKey"
+              rows={3}
+              wrap="soft"
+              value={editClusterModel.sshKey}
+              onChange={(e) => {
+                const model: ClusterModel = {
+                  ...editClusterModel,
+                  cluster: { ...editClusterModel.cluster },
+                  sshKey: e.target.value,
+                };
+                setEditClusterModel(model);
+                setReturn(model);
+              }}
+            />
+          </div>
+        </>
+      ),
+      canConfirm:
+        notEmptyOrFalsey(editClusterModel.cluster.name) &&
+        notEmptyOrFalsey(editClusterModel.cluster.description) &&
+        !notification.pending,
+    },
+    [
+      editClusterModel,
+      setEditClusterModel,
+      notification.pending,
+      editConfirmationTitle,
+    ]
+  );
+
+  const [getRemoveConfirmation] = useConfirmationDialog<string>(
+    {
+      title: "Remove Cluster",
+      message: "Are you sure you want to remove this cluster?",
+    },
+    []
+  );
+
+  useEffect(() => {
+    const fetchClusters = async () => {
+      const response = await authenticatedFetch(`/api/_/clusteradmin/clusters`);
```

Comment:
what's this strange underscore url thing?

Reply:
I tried to get it to work by making the cluster segment of the route optional `pattern: "/api/{cluster?}/{controller=Account}/{action=Index}/{id?}"`, but it would always end up returning the Home view instead of hitting the api action.

Reply:
oh funny.  alright good enough